pkg/storaged/smart-details.jsx

```diff
@@ -0,0 +1,175 @@
+    ns_removed: _("Aborted due to a removal of a namespace from the namespace inventory"),
+    aborted_format: _("Aborted due to the processing of a Format NVM command"),
+    fatal_error: _("A fatal error or unknown test error occurred while the controller was executing the device self-test operation and the operation did not complete"),
+    unknown_seg_fail: _("Completed with a segment that failed and the segment that failed is not known"),
+    known_seg_fail: _("Completed with one or more failed segments"),
+    aborted_unknown: _("Aborted for unknown reason"),
+    aborted_sanitize: _("Aborted due to a sanitize operation"),
+};
+
+// TODO: UX to show what is wrong with the nvme
+// .SmartCriticalWarning contains an array of values which say what is wrong with the disk
+// empty if disk OK
+// https://storaged.org/doc/udisks2-api/latest/gdbus-org.freedesktop.UDisks2.NVMe.Controller.html#gdbus-property-org-freedesktop-UDisks2-NVMe-Controller.SmartCriticalWarning
+const nvmeCriticalWarning = {
+    spare: _("Spare capacity is bellow the treshold"),
+    temperature: _("Temperature outside of recommended thresholds"),
+    degraded: _("Degraded"),
+    readonly: _("All media is in read-only mode"),
+    volatile_mem: _("Volatile memory backup failed"),
+    pmr_readonly: _("Persistent memory has become read-only")
+};
+
+const SmartActions = ({ drive_ata }) => {
+    const [isKebabOpen, setKebabOpen] = useState(false);
+    const smartSelftestStatus = drive_ata.SmartSelftestStatus;
+
+    const runSelfTest = (type) => {
+        drive_ata.SmartSelftestStart(type, {});
+    };
+
+    const abortSelfTest = () => {
+        drive_ata.SmartSelftestAbort({});
+    };
+
+    const actions = [
+        <DropdownItem key="smart-short-test"
+                      isDisabled={smartSelftestStatus === "inprogress"}
+                      onClick={() => { setKebabOpen(false); runSelfTest('short') }}>
+            {_("Run short test")}
+        </DropdownItem>,
+        <DropdownItem key="smart-extended-test"
+                      isDisabled={smartSelftestStatus === "inprogress"}
+                      onClick={() => { setKebabOpen(false); runSelfTest('extended') }}>
+            {_("Run extended test")}
+        </DropdownItem>,
+        <DropdownItem key="abort-smart-test"
+                      isDisabled={smartSelftestStatus !== "inprogress"}
+                      onClick={() => { setKebabOpen(false); abortSelfTest() }}>
+            {_("Abort test")}
+        </DropdownItem>,
+    ];
+
+    return (
+        <Dropdown toggle={<KebabToggle onToggle={(_, isOpen) => setKebabOpen(isOpen)} />}
+                isPlain
+                isOpen={isKebabOpen}
+                position="right"
+                id="smart-actions"
+                dropdownItems={actions}
+        />
+    );
+};
+
+export const SmartCard = ({ card, drive_ata, drive_type }) => {
+    const powerOnHours = (drive_type === "hdd")
```

Comment:
`hdd` vs `ssd` is incorrect (obviously), you need to diff it based on whether it is SATA SMART or NVMe SMART!